packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -71,7 +72,12 @@ function transform(codeGen: CodeGen): t.Expression {
         const databag = elementDataBag(element, slotParentName);
         let res: t.Expression;
 
-        if (codeGen.staticNodes.has(element) && isElement(element)) {
+        if (
+            codeGen.staticNodes.has(element) &&
+            isElement(element) &&
+            // Tags like <th> and <td> are okay as static fragments, but only if they're not at the top level
+            !isUnsafeTopLevelSerializableElement(element)
```

Comment:
I'm not a fan of the double negation like this... a more elegant approach is to have something like: `isSafeTopLevelSerializableElement(element)` instead.